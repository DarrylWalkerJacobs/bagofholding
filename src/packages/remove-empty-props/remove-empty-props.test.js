import removeEmptyProps from './remove-empty-props';

it('recursively removes `null` and `undefined` properties from payload object', () => {
  const object = {
    foo: null,
    bar: undefined,
    baz: '',
    fab: false,
    gib: '༼ つ ◕_◕ ༽つ',
    mud: {
      one: 1,
      two: '2',
      three: Number('3'),
      four: null,
      five: undefined,
      six: {},
      seven: []
    }
  };

  expect(Object.keys(removeEmptyProps(object)).length).toBe(4);
  expect(Object.keys(removeEmptyProps(object).mud).length).toBe(5);
  expect(Object.keys(removeEmptyProps(object)).length).toMatchSnapshot();
});
