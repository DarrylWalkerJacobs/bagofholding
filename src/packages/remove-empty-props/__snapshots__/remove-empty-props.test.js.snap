// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recursively removes \`null\` and \`undefined\` properties from payload object 1`] = `4`;
